biz.aQute.repository/src/aQute/bnd/repository/maven/provider/MavenBndRepository.java

```diff
@@ -194,7 +194,7 @@ public PutResult put(InputStream stream, PutOptions options) throws Exception {
 				}
 
 				logger.debug("Put release {}", pom.getRevision());
-				try (Release releaser = storage.release(pom.getRevision(), options.context.getProperties())) {
```

Comment:
Why not just use flattened properties? It is big deal to change all these API from getting a Properties object to get a Processor object. I am not sure that is a great idea.

If flattened properties do not work, then I would suggest we implement a Properties subclass which wraps a Processor and pass that.

Reply:
true that. Much easier solution!